includes/internal/emails/class-email-sender.php

```diff
@@ -147,19 +155,22 @@ public function get_email_subject( WP_Post $email_post, array $placeholders = []
 	 * @return string
 	 */
 	public function get_email_body( WP_Post $email_post, array $placeholders = [] ): string {
-		global $post;
 
-		$post = $email_post; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited -- Necessary for the post title block to work.
-		setup_postdata( $post );
+		$post_id = 'revision' === $email_post->post_type ? $email_post->post_parent : $email_post->ID;
 
-		$post_content = $this->replace_placeholders(
-			do_blocks( $email_post->post_content ),
-			$placeholders
+		// phpcs:ignore WordPress.WP.DiscouragedFunctions.query_posts_query_posts -- We need to modify the global query object in order to render templates.
+		query_posts(
+			[
+				'posts_per_page' => 1,
+				'p'              => $post_id,
+				'post_type'      => Email_Post_Type::POST_TYPE,
+			]
 		);
 
-		wp_reset_postdata();
+		the_post();
```

Comment:
Could it be a part of repository's logic?

Reply:
I dont think so because it doesnt return an entity that breaks the Repository concept.